### DISCUSSION TITLE
Multidomain support for this project

### DISCUSSION THREAD

**Discussion by richycrazy** [ltb-project/self-service-password#462] [Ideas]
First Sorry but a im not a native english
I have made some small adjustment to get this project work whit multiple domains in one single ldap server
two common methods to have a multidomain ldap are generally to use root-suffixes in the form of dc=example,dc=com and in the form o=example.com

I add another config var named $ldap_base_dn_mode = "dc";  (supported values are "dc" and "o" )
and then rewrite all the actions files in the project not all the file of course. I just replace the line where the $login variable is defined in a way that allows the users to login in the form of user@domain and the Base DN is dynamically get from the user login.
my lines of code looks like this:
```php
                    if (isset($_REQUEST["login"]) and $_REQUEST["login"]) {
                        if (isset($ldap_base_dn_mode) && $ldap_base_dn_mode ==="o") {
                            $prelogin = explode('@',strval($_REQUEST["login"]));
                            $login = $prelogin[0];
                            $ldap_base = 'o='.$prelogin[1];
                        }
                        if (isset($ldap_base_dn_mode) && $ldap_base_dn_mode === "dc") {
                            $prelogin = explode('@',strval($_REQUEST["login"]));
                            $login = $prelogin[0];
                            $dc_base = explode('.' ,$prelogin[1]);
                            for ($i=0; $i < count($dc_base) ; $i++) { 
                                if ( $i != (count($dc_base) - 1)) {
                                    $ldap_base .="dc=".$dc_base[$i].",";
                                }else {$ldap_base .= "dc=".$dc_base[$i];}
                            }
                        }
                    }
```
It will be nice you guys to support multidomain for this web.

**Comment by coudot** [comment]
Hello @richycrazy 

thanks for this suggestion.

We implemented the multi-domain support by allowing to pick a configuration depending on an HTTP header sent to SSP. See https://self-service-password.readthedocs.io/en/latest/config_general.html#multi-tenancy

It is different from your proposition but could maybe fit your needs.

**Comment by richycrazy** [comment]
Yes i think it maybe does. I have to figure it out how to automatically
generate a config file every time i add a domain. Thanks
